src/BackgroundCells.js

```diff
@@ -56,23 +57,28 @@ class BackgroundCells extends React.Component {
   }
 
   render(){
-    let { range, cellWrapperComponent: Wrapper, date: currentDate } = this.props;
+    let { range, cellWrapperComponent: Wrapper, dayPropGetter, date: currentDate } = this.props;
     let { selecting, startIdx, endIdx } = this.state;
 
     return (
       <div className='rbc-row-bg'>
         {range.map((date, index) => {
           let selected =  selecting && index >= startIdx && index <= endIdx;
+          const { className, style: dayStyles } = (dayPropGetter && dayPropGetter(date)) || {};
+          const segmStyles = segStyle(1, range.length)
+          const styles = Object.assign({}, segmStyles, dayStyles)
```

Comment:
segmentStyles should be assigned in last so they cannot be overridden by the dayPropGetter styles